### ISSUE TITLE
No such file or directory: system32\\LogFiles\\Scm\\SCM.EVM

### ISSUE THREAD

**Issue by Saibamen** [bleachbit/bleachbit#937]
`[Errno 2] No such file or directory: '\\\\?\\C:\\WINDOWS\\system32\\LogFiles\\Scm\\SCM.EVM': Command to delete C:\WINDOWS\system32\LogFiles\Scm\SCM.EVM`

BleachBit version 4.0.0

**Comment by az0** [comment]
@Saibamen 
Does this happen consistently, or did it happen just once? 

I could _not_ reproduce this error on Windows 10 with BleachBit 4.0.1.1650 portable. My steps were

1. Enabled the preference to download Winapp2.ini (where this rule comes from)
1. Restart BleachBit to download Winapp2.ini . Start it with Administrator privileges to delete files under `c:\windows\`
1. Create the directory `c:\windows\system32\logfiles\scm\` because it didn't exist on my system
1. Add some files to the directory
1. Check "Windows logs" under "Microsoft Windows"
1. Run a delete

If you know of a way to reproduce it, let me know.

**Comment by Saibamen** [comment]
Still happens in 4.0.1.1662

> Does this happen consistently, or did it happen just once?

consistently

> Add some files to the directory

This is not some file ;) I don't know what it is, but when I trying open it with Notepad++ (7.8.6), I have this window:
![image](https://user-images.githubusercontent.com/905878/85793462-5676d700-b735-11ea-92c5-e42f9e20b65f.png)
This is **not** read-only file.

While opening with standard notepad:
![image](https://user-images.githubusercontent.com/905878/85793654-a05fbd00-b735-11ea-8aae-c088ec90a2f5.png)

**Comment by az0** [comment]
Looks like a special kind of file created by Windows Service Control specific to Windows Server. I'll look into it more.

In the BleachBit preferences, do you have "overwrite files" enabled? Would you mind testing it both ways: does it make a difference?

**Comment by Saibamen** [comment]
Sorry, settings have missing change language option, so this are my settings from previous comment:
![image](https://user-images.githubusercontent.com/905878/85798796-6c3cca00-b73e-11ea-88d7-6ca780789333.png)

Checked`Nadpisz zawartość plików` and still have this error log

**Comment by az0** [comment]
Yes, "Nadpisz zawartość plików" is what I meant. Thank you for checking.

The last thing you might be able to check. Enable debugging (Wyświetl informacje o błędach), and start `bleachbit_console.exe` instead of `bleachbit.exe`. If there is a Traceback message, please post it here. (There might not be one.)

**Comment by Saibamen** [comment]
```
already an admin (UAC not required)
(auto hiding cleaners) spam log
clean_operation('firefox'), options = '['vacuum', 'cache', 'crash_reports']'
clean_operation('system'), options = '['updates', 'recycle_bin', 'logs', 'muicache', 'tmp', 'prefetch', 'memory_dump']'
[Errno 2] No such file or directory: '\\\\?\\C:\\WINDOWS\\system32\\LogFiles\\Scm\\SCM.EVM': Command to delete C:\WINDOWS\system32\LogFiles\Scm\SCM.EVM
clean_operation('google_chrome'), options = '['vacuum', 'cache']'
clean_operation('windows_defender'), options = '['logs', 'temp']'
clean_operation('internet_explorer'), options = '['logs', 'cache']'
elapsed time: 3 seconds
```

**Comment by az0** [comment]
@Saibamen 
Thank you for your patience. Would you please test with[ BleachBit version 4.2.0](https://www.bleachbit.org/news/bleachbit-420) (the latest stable release)? There were many bug fixes

**Comment by Saibamen** [comment]
@az0 This same error.
Tested with BB 4.2.0 release

**Comment by az0** [comment]
@Saibamen 
Thank you for checking, and I marked it to take a look at it for the next release.

**Comment by az0** [comment]
This could be related to WoW redirection, which affects `%windir%\system32`

**Comment by rados** [comment]
@az0 
I can handle this one. I can find this file on my Windows 7 machine but it is accessible and Bleachbit successfully cleans it. So it could be something related with a server machine indeed.

@Saibamen
Can you check what is the date of the file? Last modified, last used, etc...

**Comment by Saibamen** [comment]
![image](https://user-images.githubusercontent.com/905878/122669908-a4df7a00-d1bf-11eb-974c-3e4c50702598.png)

**Comment by az0** [comment]
After trying a few different ways, I could not make the file `logfiles\scm\scm.evm` on my Windows 11.

**Comment by az0** [comment]
@Saibamen 
Thank you for your patience.

Do you still have this error with BleachBit [CI build 5.1.0.3305-master](https://ci.bleachbit.org/?prefix=dl/5.1.0.3305-master/)? It has many changes since 4.2.0.

**Comment by az0** [comment]
The issue may be related to redirection. See https://github.com/bleachbit/bleachbit/issues/2011#issuecomment-3924686562 . I ran BleachBit from source on Windows 10 with Python 64-bit, so there was no redirection. It found the SCM files

```cmd
(bleachbit) c:\projects\bleachbit>python bleachbit.py --preview system.logs | find  /i "system32"
Delete 2.1MB C:\WINDOWS\system32\LogFiles\Scm\SCM.EVM
Delete 5.2MB C:\WINDOWS\system32\LogFiles\Scm\SCM.EVM.1
```

**Comment by az0** [comment]
I released [BleachBit version 6.0.1 beta](https://www.bleachbit.org/news/bleachbit-601-beta) with a fix. It introduces a variable `%WindowsSystem%` that points to the correct directory, depending on whether the process is 32-bit or 64-bit. It is now used for for the option **System - Logs**, **System - Updates**, and Internet Explorer (which still works on Windows 11 64-bit).

Thanks for your patience, and if you have a chance, try out version 6.0.1.

Also, BleachBit now has several ways to run a 64-bit process on Windows: [running from source](https://docs.bleachbit.org/dev/running-from-source-code.html), [text user interface (TUI)](https://www.bleachbit.org/news/bleachbit-text-user-interface), and [wxWidgets](https://ci.bleachbit.org/?prefix=dl/5.1.2.3501-wx-gui-mvp/). However, the TUI and wxWidgets are not stable releases.
